app/server/appsmith-server/src/main/java/com/appsmith/server/services/ActionCollectionServiceImpl.java

```diff
@@ -1,73 +1,338 @@
+                                       NewPageService newPageService,
+                                       PolicyGenerator policyGenerator) {
+        super(scheduler, validator, mongoConverter, reactiveMongoTemplate, repository, analyticsService);
         this.collectionService = collectionService;
         this.layoutActionService = layoutActionService;
         this.newActionService = newActionService;
+        this.newPageService = newPageService;
+        this.policyGenerator = policyGenerator;
+    }
+
+    @Override
+    public Flux<ActionCollection> findAllByApplicationIdAndViewMode(String applicationId, Boolean viewMode, AclPermission permission, Sort sort) {
+        return repository.findByApplicationIdAndViewMode(applicationId, viewMode, permission)
+                // In case of view mode being true, filter out all the actions which haven't been published
+                .flatMap(collection -> {
+                    if (Boolean.TRUE.equals(viewMode)) {
+                        // In case we are trying to fetch published actions but this action has not been published, do not return
+                        if (collection.getPublishedCollection() == null) {
+                            return Mono.empty();
+                        }
+                    }
+                    // No need to handle the edge case of unpublished action not being present. This is not possible because
+                    // every created action starts from an unpublishedAction state.
+
+                    return Mono.just(collection);
+                });
     }
 
     /**
-     * Called by Collection controller to create Collection.
+     * Called by ActionCollection controller to create ActionCollection.
      *
      * @param collection
      * @return
      */
     @Override
-    public Mono<Collection> createCollection(Collection collection) {
+    public Mono<ActionCollectionDTO> createCollection(ActionCollectionDTO collection) {
         if (collection.getId() != null) {
-            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, "id"));
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.ID));
         }
 
-        if (collection.getActions() == null) {
-            return collectionService.create(collection);
+        final Set<String> validationMessages = collection.validate();
+        if (!validationMessages.isEmpty()) {
+            return Mono.error(new AppsmithException(
+                    AppsmithError.INVALID_ACTION_COLLECTION,
+                    collection.getName(),
+                    validationMessages.toString()));
         }
 
-        return Mono.just(collection.getActions())
+        final String pageId = collection.getPageId();
+        MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add(FieldName.PAGE_ID, pageId);
+        Mono<NewPage> pageMono = newPageService
+                .findById(pageId, READ_PAGES)
```

Comment:
Since this is an edit event, we should be fetching pages with manage permissions instead.
```suggestion
                .findById(pageId, MANAGE_PAGES)
```